src/main/scala/SearchTweets.scala

```diff
@@ -16,24 +16,49 @@ import java.io.IOException;
 import java.net.URISyntaxException;
 import scala.collection.mutable.ArrayBuffer
 
+import scala.util.parsing.json._
 
 object SearchTweets {
 
   @throws[IOException]
   def main(args: Array[String]): Unit = {
 
-    var bearerToken: String = "Sundoo"
+    var bearerToken: String = "YOUR API KEY HERE"
 
     if (bearerToken != null) {
           Future {
 
-      var response: String = getTweets("BurgerKing", bearerToken)
-      println(response)
+            var response: String = getTweets("BurgerKing", bearerToken)
+            println(response)
+            //deconstruct JSON object to find next_token
+            val json:Option[Any] = JSON.parseFull(response)
+            val map:Map[String,Any] = json.get.asInstanceOf[Map[String, Any]]
+            val meta:Map[String, String] = map.get("meta").get.asInstanceOf[Map[String, String]]
+            //if there is next_token, call retrievePages function to recursively retrieve all pages
+            if (meta.contains("next_token")) {
+              val next_token = meta("next_token")
+              retrievePages(next_token)
+            }
           }
     } else {
       println("There was a problem getting you bearer token. Please make sure you set the BEARER_TOKEN environment variable")
     }
-    Thread.sleep(3000)
+    Thread.sleep(10000)
```

Comment:
Is it possible to change this to an 
```scala
    Await.ready(YOUR_FUTURE_HERE, 10000)
```

Reply:
Yes I could. The code as is is copied from the twitter sample code where they use sleeps. Is it better to use Await.ready? Was also wondering if there is a way to do it/a method where you don't have to specify a sleeping time but not sure if that exists.

Reply:
Await.ready makes it sleep until either the future is finished or the given time is reached so its about as close to that as you want to get

Reply:
That sounds good. I was also looking at .onComplete() for handling futures- do you know if that one requires a sleep()?

Reply:
onComplete is more for passing a function that runs after the future is ran while Await.ready is closer to a sleep